level-2/주차-요금-계산.js

```diff
@@ -0,0 +1,61 @@
+function solution(fees, records) {
+  const answer = []
+
+  const parkingTable = getParkingTable(records)
+  const sortedKeys = Object.keys(parkingTable).sort()
+
+  for (const targetCar of sortedKeys) {
+    if (parkingTable[targetCar].length % 2 === 1) {
+      parkingTable[targetCar].push(['23:59', 'OUT'])
+    }
+    const totalTime = getTotalTime(parkingTable, targetCar)
+    const fee = calculateFee(totalTime, fees)
+    answer.push(fee)
+  }
+
+  return answer
+}
+
+function getParkingTable(records) {
+  const parkingTable = {}
+
+  records.forEach((val) => {
+    const info = val.split(' ')
+    if (Object.keys(parkingTable).includes(info[1])) {
+      parkingTable[info[1]].push([info[0], info[2]])
+    } else {
+      parkingTable[info[1]] = [[info[0], info[2]]]
+    }
+  })
+
+  return parkingTable
+}
+
+function getTotalTime(parkingTable, targetCar) {
+  let totalTime = 0
+  for (let i = 0; i < parkingTable[targetCar].length - 1; i += 2) {
+    totalTime += calculateTotalTime(
+      parkingTable[targetCar][i][0],
+      parkingTable[targetCar][i + 1][0]
+    )
+  }
+  return totalTime
+}
+
+function calculateTotalTime(time1, time2) {
+  const [hour1, minute1] = time1.split(':').map((val) => parseInt(val))
+  const [hour2, minute2] = time2.split(':').map((val) => parseInt(val))
+
+  return hour2 * 60 + minute2 - (hour1 * 60 + minute1)
+}
+
+function calculateFee(totalTime, feeInfo) {
+  let fee = 0
+  if (feeInfo[0] >= totalTime) {
+    fee = feeInfo[1]
+  } else {
+    fee =
+      feeInfo[1] + Math.ceil((totalTime - feeInfo[0]) / feeInfo[2]) * feeInfo[3]
+  }
+  return fee
```

Comment:
🌱 반영해도 좋고 넘어가도 좋아요. 🌱

변수에 값을 담아 바로 return 을 하는 형식이기 때문에, 불필요한 변수 선언이나 else 문 없이 early return 패턴을 사용하시면 더 깔끔한 코드가 될 것 같다고 생각합니다!
[참고링크](https://gomjellie.github.io/%ED%8C%A8%ED%84%B4/2020/04/27/early-return.html)

```suggestion
  if (feeInfo[0] >= totalTime) return feeInfo[1]
  return feeInfo[1] + Math.ceil((totalTime - feeInfo[0]) / feeInfo[2]) * feeInfo[3]
```